Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs

```diff
@@ -31,19 +33,47 @@ public class MixedRealityManager : MonoBehaviour
         public MixedRealityConfigurationProfile ActiveProfile
         {
             get { return activeProfile; }
-            set { activeProfile = value; ResetConfiguration(); }
+            set { activeProfile = value; if (resetOnProfileChange) gameObject.SetActive(true); ResetConfiguration(); }
         }
 
+        /// <summary>
+        /// When a configuration Profile is replaced with a new configuration, force all managers to reset and read the new values
+        /// </summary>
         private void ResetConfiguration()
         {
+            // Reset all active managers in the registry
             foreach (var manager in ActiveProfile.ActiveManagers)
             {
                 manager.Value.Reset();
             }
+
+            // Reset all registered runtime components
+            foreach (var manager in MixedRealityComponents)
+            {
+                manager.Item2.Reset();
```

Comment:
What's Item2?

Reply:
Tuple

Reply:
Can we just name them as they are?

Reply:
Or is this name hard coded in the pattern?